### ISSUE TITLE
How to activate the plugin für groovy or javascript only projects

### ISSUE THREAD

**Issue by Janpopan** [porscheinformatik/sonarqube-licensecheck#182]
the plugin is only worky if you have at least on java file in your setup.

We have JavaScript and also groovy only projects - in this case you can't activate the plugin.

Thanks for the plugin, great work.

**Comment by derhecht** [comment]
use the sonar scanner? which at least you need to have Sonar results at all or do you use a javascript sonar plugin?

**Comment by Janpopan** [comment]
There are only java rules which activate the plugin. And the second, the javascanner is only activated in the case, if a java file is found. So javascript only projects are "unsupported".

**Comment by derhecht** [comment]
the scanner is initiated with 

this.scanners = new Scanner[]{
            new PackageJsonDependencyScanner(
                configuration.getBoolean(LicenseCheckPropertyKeys.NPM_RESOLVE_TRANSITIVE_DEPS).orElse(false)),
            new MavenDependencyScanner(mavenLicenseService, mavenDependencyService),
            new GradleDependencyScanner(mavenLicenseService)};

so it will resolve in a list containing dependencies for all Maven, Gradle and package.json found

it is correct, that atm only java rules are setup but in the end the licenses at all should be visible in regarding project

**Comment by derhecht** [comment]
what do you mean by a "Groovy only" project? it is build with gradle? so where the plugin should get dependency information from?

**Comment by Janpopan** [comment]
As I said, in a pure js project you have to add At least one java file to add in the project, to activate the rule, to See tue Result 
If you look in a Future and you Think about using an vorher license source maybe pivotal, the current solution is a strong limitation.

**Comment by Janpopan** [comment]
> what do you mean by a "Groovy only" project? it is build with gradle? so where the plugin should get dependency information from?

Ist the same like js you nerd AT least one java file.
I Think the plugin should change the activation process more like the owasp dependency plugin

**Comment by vladislavmungalov** [comment]
I've added one empty java file to a project folder, but the plugin doesn't work for package.json dependencies (using TypeScript project). 
Moreover I've added TypeScript to LicenseCheckRulesDefinition.java from feature/182 branch, that doesn't work too.
Any ideas?

**Comment by Janpopan** [comment]
Did you enable the Java rules for this plugin in your java Profile in sonarqube and did sonar show you java as a language in your project

**Comment by vladislavmungalov** [comment]
@Janpopan I have enabled the Java rules, the plugin works fine with Maven Java projects. 
Sonar doesn't show me Java as a language, only TypeScript, HTML, CSS, HTML, XML. 
I've created an empty .java file in the project folder without POM file. 
Moreover I have a warning from sonar: 
> Bytecode of dependencies was not provided for analysis of source files, you might end up with less precise results. Bytecode can be provided using sonar.java.libraries property.

**Comment by vladislavmungalov** [comment]
@Janpopan P.S. package.json is not in the project root folder, but in another package of the root folder.

**Comment by vladislavmungalov** [comment]
@Janpopan I've tried it with another JavaScript project, that's a standard project, package.json is in the project root folder. 

If I just create an empty Java file in the root folder, then I have a warning from sonarqube:
> Bytecode of dependencies was not provided for analysis of source files, you might end up with less precise results. Bytecode can be provided using sonar.java.libraries property.

If I copy pom.xml from another project and change some properties, then I have one more warning:
> Missing blame information for 1 file. This may lead to some features not working correctly. Please check the analysis logs.

**Comment by Janpopan** [comment]
Did you see the java scanner in the sonar Client log?

**Comment by vladislavmungalov** [comment]
This workaround worked for me:
1) Create an empty java file in [project_root_folder]/src/main/java/com/company/project/File.java
This path is important, so that your project can be interpreted as a Java project.
2) npm install
3) run sonar

@Janpopan thank you for the help

**Comment by ricardojdsilva87** [comment]
Hello we encountered a similar problem, in this case with the CSS, Typescript and HTML profiles activated
```
INFO: Sensor License Check [licensecheck]
INFO: Scanning for NPM dependencies
INFO: Scanning NPM packages [xxxx, xxxxx, xxxxx, xxxxx]
INFO: No pom.xml file found in /home/jenkins/agent/workspace/project_path - skipping Maven dependency scan
INFO: No license-details.json file found in /home/jenkins/agent/project_path/build/reports/dependency-license/license-details.json - skipping Gradle dependency scan
INFO: No License found for Dependency xxxxx
INFO: Dependency @stencil/router uses a not allowed license MIT License 
INFO: Dependency @types/node uses a not allowed license MIT License 
INFO: Dependency js-cookie uses a not allowed license MIT License 
INFO: Dependency oidc-client uses a not allowed license Apache License 2.0 
INFO: Dependency simple-git uses a not allowed license MIT License 
INFO: No License found for Dependency xxxxxx
INFO: Sensor License Check [licensecheck] (done) | time=42ms
```

It seems that altough the license-checker runs with no issues, the results are not being shown on sonarqube.
When adding the needed rules on the quality profile, also only the ones for Java appear.

Is there some extra step needed to do to show the licenses on sonarqube?
Thanks for the help

**Comment by derhecht** [comment]
it is not supported atm - may review and approve regarding PR to have this feature enabled

**Comment by axi92** [comment]
> This workaround worked for me:
> 
>     1. Create a java file in [project_root_folder]/src/main/java/com/company/project/File.java with package name from the path:
>        package src.main.java.com.company.project;
>        This path is important, so that your project can be interpreted as a Java project.
> 
>     2. npm install
> 
>     3. run sonar
> 
> 
> @Janpopan thank you for the help

Is the only content of that File.java this?
`package src.main.java.com.company.project;`
I tried it like this but got this:
```
WARN: Missing blame information for the following files:
WARN:   * src/main/java/com/company/project/File.java
WARN: This may lead to missing/broken features in SonarQube
```

**Comment by oliverlockwood** [comment]
@Janpopan unfortunately the workaround becomes a bit of a rabbit hole.  For my NodeJS project:

- I create a temporary dummy Java file as part of the CI build process, right before running Sonar.
- I ensure that I have a Java Sonar quality profile which includes the two `licensecheck` rules, and nothing else, and assign my project to this quality profile.  See screenshot (note I have redacted the project names to anonymise this)
![Screenshot 2021-03-02 at 10 16 15](https://user-images.githubusercontent.com/6017680/109633845-7b331200-7b40-11eb-9ab6-b1e7ba305a70.png)
- Since I have real Java projects, I use FindBugs etc on them; you can't disable the FindBugs exporter on a per-profile basis, even if you have no FindBugs rules
- FindBugs complains that my Java file is not compiled and crashes, meaning there is no Sonar output:

```
ERROR: Error during SonarQube Scanner execution
java.lang.IllegalStateException: Can not execute Findbugs
	at org.sonar.plugins.findbugs.FindbugsExecutor.execute(FindbugsExecutor.java:188)
	at org.sonar.plugins.findbugs.FindbugsSensor.execute(FindbugsSensor.java:114)
	at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:48)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:85)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.lambda$execute$1(ModuleSensorsExecutor.java:59)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.withModuleStrategy(ModuleSensorsExecutor.java:77)
	at org.sonar.scanner.sensor.ModuleSensorsExecutor.execute(ModuleSensorsExecutor.java:59)
	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:82)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:137)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:123)
	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:393)
	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:389)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:358)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:137)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:123)
	at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:144)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:137)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:123)
	at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:72)
	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:66)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:185)
	at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:137)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:112)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:75)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
Caused by: java.lang.IllegalStateException: One (sub)project contains Java source files that are not compiled (/var/lib/jenkins/workspace/REDACTED).
	at org.sonar.plugins.findbugs.FindbugsConfiguration.getFindbugsProject(FindbugsConfiguration.java:123)
	at org.sonar.plugins.findbugs.FindbugsExecutor.execute(FindbugsExecutor.java:119)
	... 31 more
```

I'm looking into inserting _compiled_ versions of the dummy Java code in my NodeJS project as well, but this is fast becoming a complete rabbit hole.

Can't we instead change the definition of the licensecheck rules defined in the plugin, so they can be applied to non-Java projects?  That seems like the correct thing to do, rather than this painful workaround.

**Comment by derkoe** [comment]
Feature released in [v5.0.0-beta-1](https://github.com/porscheinformatik/sonarqube-licensecheck/releases/tag/v5.0.0-beta-1)
